src/main/java/seedu/address/model/recipe/Portion.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.model.recipe;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Objects;
+/**
+ * Represents a Recipe's portion in the recipe book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPortion(double)}
+ */
+public class Portion {
+    public static final String MESSAGE_CONSTRAINTS = "Portion size should not be left blank";
+
+    public final double value;
+
+    /**
+     * Constructs a {@code Portion}.
+     *
+     * @param portion A valid portion size.
+     */
+    public Portion(double portion) {
+        checkArgument(isValidPortion(portion), MESSAGE_CONSTRAINTS);
+        value = portion;
+    }
```

Comment:
Think there's a missing requireNonNull check here? Constructor in the Name class is as follows:

```
public Name(String name) {
        requireNonNull(name);
        checkArgument(isValidName(name), MESSAGE_CONSTRAINTS);
        fullName = name;
    }
```

Maybe we should add the non-null check as well?

Reply:
this one should be fine though, if value isnt specified, portion would be 0 by default, assuming it can get to this point without being specified

Reply:
Yeap, as Yusuf have mentioned, performing requireNonNull on a primitive type is redundant because null for integers is 0.